src/compiler/checker.ts

```diff
@@ -40298,7 +40381,9 @@ namespace ts {
         }
 
         function getNodeCheckFlags(node: Node): NodeCheckFlags {
-            return getNodeLinks(node).flags || 0;
+            const nodeId = node.id || 0;
+            if (nodeId < 0 || nodeId >= nodeLinks.length) return 0;
+            return nodeLinks[nodeId]?.flags || 0;

```

Comment:
this should return 0 even for the case that `nodeId < 0 || nodeId >= nodeLinks.length`, right? Is the previous line needed if so?

Reply:
The boundary check isn't strictly necessary, but boundary checked indexed access on arrays in V8 is faster than unchecked (since `-1` becomes a property lookup, not an array access).